### DISCUSSION TITLE
Store and Forward

### DISCUSSION THREAD

**Discussion by galvezmjorge** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#24] [Q&A]
Hello! I would like to enable Store and Forward but the documentation has not information about how to set it up. Have you tried to make this feature work? Thanks!

**Comment by thomassorensen2650** [comment]
the store-forward functionality could definitely use some more documentation. 

The device node will buffer and store forward data (DDATA) by default, when it loses connection to the MQTT broker. Commands will not be store-forwarded for obvious reasons. 

The additional store-forward functionality that can be enabled is using the Sparkplug B Primary host functionality to buffer in case the MQTT broker is available, but the "primary host" is not connected to the broker. See page 16  [in the Sparkplug Specification](https://sparkplug.eclipse.org/specification/version/3.0/documents/sparkplug-specification-3.0.0.pdf) for more information about how this works. Basically the destination for the data needs to support this functionality and publish its ONLINE state to the broker.

The data will be buffered in the default node-red context store. This is by default in-memory buffer, but can be disk backed, by [modifying the default node-red context configuration](https://nodered.org/docs/user-guide/context)

**Reply by Theodorska1** [reply]
"Basically the destination for the data needs to support this functionality and publish its ONLINE state to the broker."

What settings are required to enable this to work with Ignition? I am getting destination offline, I have tried to use the SCADA name, SERVER name, Historian server name, etc.

**Reply by Theodorska1** [reply]
I think I found it as the primary host ID in Config>Mqttengine>MQTT Engine Settings at our ignition server
